packages/smooth_app/pubspec.yaml

```diff
@@ -42,7 +42,7 @@ dependencies:
   photo_view: ^0.14.0
   uuid: ^3.0.6
   provider: ^6.0.3
-  sentry_flutter: ^6.6.0 # careful with upgrading cf: https://github.com/openfoodfacts/smooth-app/issues/1300
+  sentry_flutter: ^6.6.3 # careful with upgrading cf: https://github.com/openfoodfacts/smooth-app/issues/1300
```

Comment:
What's it use case for this comment now, just curious as we already are updating it to the latest version of sentry
So I think this comment is not longer useful ? Is it

Reply:
@AshAman999 Just have a look at the link: there are side-effects on iOS each time we upgrade the sentry version.
Therefore, we need to be careful. Always.

Reply:
@AshAman999

<details>

<summary>Something like that, that's just happened</summary>

```
% flutter run
"yi": 425 untranslated message(s).
To see a detailed report, use the untranslated-messages-file 
option in the l10n.yaml file:
untranslated-messages-file: desiredFileName.txt
<other option>: <other selection> 


This will generate a JSON format file containing all messages that 
need to be translated.
Launching lib/main.dart on iPhone 8 Plus in debug mode...
Running pod install...                                              8,9s
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)

    Finding Podfile changes
      - Flutter
      - audioplayers_darwin
      - camera
      - data_importer
      - device_info_plus
      - flutter_email_sender
      - flutter_isolate
      - flutter_native_splash
      - flutter_secure_storage
      - google_mlkit_barcode_scanning
      - google_mlkit_commons
      - image_cropper
      - image_picker_ios
      - integration_test
      - iso_countries
      - package_info_plus
      - path_provider_ios
      - permission_handler_apple
      - sentry_flutter
      - share_plus
      - shared_preferences_ios
      - sqflite
      - url_launcher_ios

    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `audioplayers_darwin` from
    `.symlinks/plugins/audioplayers_darwin/ios`
    -> Fetching podspec for `camera` from `.symlinks/plugins/camera/ios`
    -> Fetching podspec for `data_importer` from
    `.symlinks/plugins/data_importer/ios`
    -> Fetching podspec for `device_info_plus` from
    `.symlinks/plugins/device_info_plus/ios`
    -> Fetching podspec for `flutter_email_sender` from
    `.symlinks/plugins/flutter_email_sender/ios`
    -> Fetching podspec for `flutter_isolate` from
    `.symlinks/plugins/flutter_isolate/ios`
    -> Fetching podspec for `flutter_native_splash` from
    `.symlinks/plugins/flutter_native_splash/ios`
    -> Fetching podspec for `flutter_secure_storage` from
    `.symlinks/plugins/flutter_secure_storage/ios`
    -> Fetching podspec for `google_mlkit_barcode_scanning` from
    `.symlinks/plugins/google_mlkit_barcode_scanning/ios`
    -> Fetching podspec for `google_mlkit_commons` from
    `.symlinks/plugins/google_mlkit_commons/ios`
    -> Fetching podspec for `image_cropper` from
    `.symlinks/plugins/image_cropper/ios`
    -> Fetching podspec for `image_picker_ios` from
    `.symlinks/plugins/image_picker_ios/ios`
    -> Fetching podspec for `integration_test` from
    `.symlinks/plugins/integration_test/ios`
    -> Fetching podspec for `iso_countries` from
    `.symlinks/plugins/iso_countries/ios`
    -> Fetching podspec for `package_info_plus` from
    `.symlinks/plugins/package_info_plus/ios`
    -> Fetching podspec for `path_provider_ios` from
    `.symlinks/plugins/path_provider_ios/ios`
    -> Fetching podspec for `permission_handler_apple` from
    `.symlinks/plugins/permission_handler_apple/ios`
    -> Fetching podspec for `sentry_flutter` from
    `.symlinks/plugins/sentry_flutter/ios`
    -> Fetching podspec for `share_plus` from `.symlinks/plugins/share_plus/ios`
    -> Fetching podspec for `shared_preferences_ios` from
    `.symlinks/plugins/shared_preferences_ios/ios`
    -> Fetching podspec for `sqflite` from `.symlinks/plugins/sqflite/ios`
    -> Fetching podspec for `url_launcher_ios` from
    `.symlinks/plugins/url_launcher_ios/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local
      because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_f_6_3.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/f/6/3/KeychainAccess/4.2.2/KeychainAccess.podspec.json exists!
      Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_9_4_5.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/9/4/5/RealmSwift/10.28.1/RealmSwift.podspec.json exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_b_e_b.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/b/e/b/GoogleMLKit/3.0.0/GoogleMLKit.podspec.json exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/b/e/b/GoogleMLKit/2.6.0/GoogleMLKit.podspec.json exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_8_1_e.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/8/1/e/MLKitVision/4.0.0/MLKitVision.podspec.json exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_a_b_d.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/a/b/d/Sentry/7.18.0/Sentry.podspec.json
      exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_4_2_c.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/4/2/c/FlutterMacOS/3.0.0/FlutterMacOS.podspec.json exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_f_4_e.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/f/4/e/FMDB/2.7.5/FMDB.podspec.json exists!
      Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_3_7_4.txt exists! Returning
      local because checking is only performed in repo update
      CDN: trunk Relative path:
      Specs/3/7/4/TOCropViewController/2.6.1/TOCropViewController.podspec.json
      exists! Returning local because checking is only performed in repo update
    [!] CocoaPods could not find compatible versions for pod "Sentry":
      In snapshot (Podfile.lock):
        Sentry (= 7.18.0, ~> 7.18.0)

      In Podfile:
        sentry_flutter (from `.symlinks/plugins/sentry_flutter/ios`) was
        resolved to 0.0.1, which depends on
          Sentry (~> 7.19.0)

    Specs satisfying the `Sentry (= 7.18.0, ~> 7.18.0), Sentry (~> 7.19.0)`
    dependency were found, but they required a higher minimum deployment target.

    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:31
    7:in `raise_error_unless_state'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:29
    9:in `block in unwind_for_conflict'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:29
    7:in `tap'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:29
    7:in `unwind_for_conflict'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:25
    7:in `process_topmost_state'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:18
    2:in `resolve'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolver.rb:43:i
    n `resolve'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/resolver.rb:94:
    in `resolve'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer/analy
    zer.rb:1078:in `block in resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/user_interface.
    rb:64:in `section'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer/analy
    zer.rb:1076:in `resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer/analy
    zer.rb:124:in `analyze'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:41
    6:in `analyze'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:24
    1:in `block in resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/user_interface.
    rb:64:in `section'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:24
    0:in `resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:16
    1:in `install!'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/command/install
    .rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:334:in
    `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/command.rb:52:i
    n `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.11.2/bin/pod:55:in `<top
    (required)>'
    /usr/local/bin/pod:23:in `load'
    /usr/local/bin/pod:23:in `<main>'

Error running pod install
Error launching application on iPhone 8 Plus.
```
</details>

Reply:
And sometimes it just stops working without upgrading :(

Reply:
Well actually we should put `6.6.3` and not `^6.6.3` if we wanted to be safe, upgrade-wise.

Reply:
Had a look and understood why it's supposed to be there
Thanks for the explaination @monsieurtanuki @M123-dev  :)